mitmproxy/tools/web/app.py

```diff
@@ -34,6 +34,12 @@ def flow_to_json(flow: mitmproxy.flow.Flow) -> dict:
         "type": flow.type,
         "modified": flow.modified(),
     }
+    # .alpn_proto_negotiated is bytes, we need to decode that.
+    for conn in "client_conn", "server_conn":
+        if f[conn]["alpn_proto_negotiated"] is None:
+            continue
+        f[conn]["alpn_proto_negotiated"] = \
+            f[conn]["alpn_proto_negotiated"].decode(errors="backslashreplace")
```

Comment:
If you look in the ALPN RFC they state that the protocol identifier may be utf8, but it is only specified as a bytestring. My understanding is that it could be (in theory, at least) a random sequence of bytes, which we cannot necessarily decode.